src/main/scala/cromwell/binding/types/WdlType.scala

```diff
@@ -54,6 +56,32 @@ trait WdlType {
 
     ast.wdlValue(this, wdlSyntaxErrorFormatter)
   }
+
+  def invalid(operation: String) = Failure(new WdlExpressionException(s"Cannot perform operation: $operation"))
+  def add(rhs: WdlType): Try[WdlType] = invalid(s"$this + $rhs")
+  def subtract(rhs: WdlType): Try[WdlType] = invalid(s"$this - $rhs")
+  def multiply(rhs: WdlType): Try[WdlType] = invalid(s"$this * $rhs")
+  def divide(rhs: WdlType): Try[WdlType] = invalid(s"$this / $rhs")
+  def mod(rhs: WdlType): Try[WdlType] = invalid(s"$this % $rhs")
+  def equals(rhs: WdlType): Try[WdlType] = invalid(s"$this == $rhs")
```

Comment:
Actually, I may have misinterpreted you... I'm not sure if this can be defined statically for all WdlTypes if that's what you're asking.  I'll think it over